python/pyspark/shuffle.py

```diff
@@ -796,7 +796,7 @@ def load_partition(j):
 
         if self._sorted:
             # all the partitions are already sorted
-            sorted_items = heapq.merge(disk_items, key=operator.itemgetter(0))
+            sorted_items = heapq.merge(*disk_items, key=operator.itemgetter(0))
```

Comment:
I missed one place here which caused the test failure. I double checked this is the last one I missed.